src/main/java/kitchenpos/domain/Order.java

```diff
@@ -10,43 +13,88 @@ public class Order {
+        this.orderLineItems = orderLineItems;
     }
 
-    public void setId(final Long id) {
+    public Order(Long id, Long orderTableId, String orderStatus, LocalDateTime orderedTime) {
         this.id = id;
+        this.orderTableId = orderTableId;
+        this.orderStatus = orderStatus;
+        this.orderedTime = orderedTime;
     }
 
-    public Long getOrderTableId() {
-        return orderTableId;
+    public Order(Long id, Long orderTableId, String orderStatus, LocalDateTime orderedTime,
+                 List<OrderLineItem> orderLineItems) {
+        this.id = id;
+        this.orderTableId = orderTableId;
+        this.orderStatus = orderStatus;
+        this.orderedTime = orderedTime;
+        this.orderLineItems = orderLineItems;
     }
 
-    public void setOrderTableId(final Long orderTableId) {
+    public Order(Long orderTableId, String orderStatus,
+                 List<OrderLineItem> orderLineItems) {
+        validateOrderLineItems(orderLineItems);
         this.orderTableId = orderTableId;
+        this.orderStatus = orderStatus;
+        this.orderLineItems = orderLineItems;
     }
 
-    public String getOrderStatus() {
-        return orderStatus;
+    private void validateOrderLineItems(List<OrderLineItem> orderLineItems) {
+        if (CollectionUtils.isEmpty(orderLineItems)) {
+            throw new IllegalArgumentException();
+        }
     }
 
-    public void setOrderStatus(final String orderStatus) {
-        this.orderStatus = orderStatus;
+    public Long getId() {
+        return id;
     }
 
-    public LocalDateTime getOrderedTime() {
-        return orderedTime;
+    public Long getOrderTableId() {
+        return orderTableId;
     }
 
-    public void setOrderedTime(final LocalDateTime orderedTime) {
-        this.orderedTime = orderedTime;
+    public String getOrderStatus() {
+        return orderStatus;
+    }
+
+    public LocalDateTime getOrderedTime() {
+        return orderedTime;
     }
 
     public List<OrderLineItem> getOrderLineItems() {
         return orderLineItems;
     }
 
-    public void setOrderLineItems(final List<OrderLineItem> orderLineItems) {
+    public void changeOrderLineItems(final List<OrderLineItem> orderLineItems) {
         this.orderLineItems = orderLineItems;
     }
+
+    public List<Long> getOrderMenuIds() {
+        return orderLineItems.stream()
+                .map(OrderLineItem::getMenuId)
+                .collect(Collectors.toList());
+    }
+
+    public void validateOrderLineItemsSize(long validSize) {
+        if (orderLineItems.size() != validSize) {
+            throw new IllegalArgumentException();
+        }
+    }
+
+    public void validateOrderStatus() {
+        if (Objects.equals(OrderStatus.COMPLETION.name(), orderStatus)) {
+            throw new IllegalArgumentException();
+        }
+    }
+
+    public void changeOrderStatus(Order order) {
+        final OrderStatus orderStatus = OrderStatus.valueOf(order.getOrderStatus());
```

Comment:
```suggestion
        final OrderStatus orderStatus = OrderStatus.valueOf(order.orderStatus);
```
로 getter가 필요 없겠네요! 근데, 파라미터에 Order말고 String이나 OrderStatus를 받을 수도 있을 것 같아요!!

Reply:
넵 변경했습니다!